_pytest/logging.py

```diff
@@ -371,18 +371,20 @@ def _runtest_for(self, item, when):
                            formatter=self.formatter, level=self.log_level) as log_handler:
             if self.log_cli_handler:
                 self.log_cli_handler.set_when(when)
```

Comment:
I would add the following code here and remove your `if item` checks:

```py
if item is None:
    yield
    return
```

Reply:
Good point.